interp/interp_test.go

```diff
@@ -191,3 +190,465 @@ func TestPiecewiseConstantPredict(t *testing.T) {
+		},
+		{
+			xs:    []float64{0},
+			ys:    []float64{0},
+			dydxs: []float64{0},
+		},
+		{
+			xs:    []float64{0, 1, 1},
+			ys:    []float64{10, 20, 10},
+			dydxs: []float64{0, 0, 0},
+		},
+	} {
+		var pc PiecewiseCubic
+		if !panics(func() { pc.FitWithDerivatives(test.xs, test.ys, test.dydxs) }) {
+			t.Errorf("expected panic for xs: %v, ys: %v and dydxs: %v", test.xs, test.ys, test.dydxs)
+		}
+	}
+}
+
+func TestAkimaSpline(t *testing.T) {
+	t.Parallel()
+	const (
+		nPts = 40
+		h    = 1e-8
+		tol  = 1e-14
+	)
+	for i, test := range []struct {
+		xs []float64
+		f  func(float64) float64
+	}{
+		{
+			xs: []float64{-5, -3, -2, -1.5, -1, 0.5, 1.5, 2.5, 3},
+			f:  func(x float64) float64 { return x * x },
+		},
+		{
+			xs: []float64{-5, -3, -2, -1.5, -1, 0.5, 1.5, 2.5, 3},
+			f:  func(x float64) float64 { return math.Pow(x, 3.) - x*x + 2 },
+		},
+		{
+			xs: []float64{-5, -3, -2, -1.5, -1, 0.5, 1.5, 2.5, 3},
+			f:  func(x float64) float64 { return -10 * x },
+		},
+		{
+			xs: []float64{-5, -3, -2, -1.5, -1, 0.5, 1.5, 2.5, 3},
+			f:  math.Sin,
+		},
+		{
+			xs: []float64{0, 1},
+			f:  math.Exp,
+		},
+		{
+			xs: []float64{-1, 0.5},
+			f:  math.Cos,
+		},
+	} {
+		var as AkimaSpline
+		n := len(test.xs)
+		ys := applyFunc(test.xs, test.f)
+		err := as.Fit(test.xs, ys)
+		if err != nil {
+			t.Errorf("Error when fitting AkimaSpline in test case %d: %v", i, err)
+		}
+		for j := 0; j < n; j++ {
+			x := test.xs[j]
+			got := as.Predict(x)
+			want := test.f(x)
+			if math.Abs(got-want) > tol {
+				t.Errorf("Mismatch in interpolated value at x == %g for test case %d: got %v, want %g", x, i, got, want)
+			}
+		}
+		if n == 2 {
+			got := as.cubic.coeffs.At(0, 1)
+			want := (ys[1] - ys[0]) / (test.xs[1] - test.xs[0])
+			if math.Abs(got-want) > tol {
+				t.Errorf("Mismatch in approximated slope for length-2 test case %d: got %v, want %g", i, got, want)
+			}
+			for j := 2; i < 4; j++ {
+				got := as.cubic.coeffs.At(0, j)
+				if got != 0 {
+					t.Errorf("Non-zero order-%d coefficient for length-2 test case %d: got %v", j, i, got)
+				}
+			}
+		}
+	}
+}
+
+func TestAkimaSplineFitErrors(t *testing.T) {
+	t.Parallel()
+	for _, test := range []struct {
+		xs, ys []float64
+	}{
+		{
+			xs: []float64{0, 1, 2},
+			ys: []float64{10, 20},
+		},
+		{
+			xs: []float64{0, 1},
+			ys: []float64{10, 20, 30},
+		},
+		{
+			xs: []float64{0},
+			ys: []float64{0},
+		},
+		{
+			xs: []float64{0, 1, 1},
+			ys: []float64{10, 20, 10},
+		},
+		{
+			xs: []float64{0, 2, 1},
+			ys: []float64{10, 20, 10},
+		},
+		{
+			xs: []float64{0, 0},
+			ys: []float64{-1, 2},
+		},
+		{
+			xs: []float64{0, -1},
+			ys: []float64{-1, 2},
+		},
+	} {
+		var as AkimaSpline
+		if !panics(func() { _ = as.Fit(test.xs, test.ys) }) {
+			t.Errorf("expected panic for xs: %v and ys: %v", test.xs, test.ys)
+		}
+	}
+}
+
+func TestAkimaWeightedAverage(t *testing.T) {
+	t.Parallel()
+	for i, test := range []struct {
+		v1, v2, w1, w2, want float64
```

Comment:
Please add a comment saying where the `want` results come from here and in the other Akima tests.